### PR TITLE
Fix examples 02 and 03 for CUDA backend

### PR DESCRIPTION
**Pull request by pkestene**
This fix issue #58 (make sure examples 02 and 03 can run when Kokkos::Cuda backend is activated)

- both examples were dereferencing a host pointer inside a device kernel
- in example 03, a Kokkos fence was missing; needed to be sure that results data can be safely re-used on host.

### CODE REVIEW

**Review by dalg24 — APPROVED** [review]

File: examples/03-axpy-view/axpy-view-cxx.cc [hunk 1]
```diff
@@ -48,13 +48,17 @@ extern "C" {
     view_type y = **v_y;
     view_type x = **v_x;
 
+    double d_alpha = *alpha;
     Kokkos::parallel_for( "axpy", y.extent(0), KOKKOS_LAMBDA( const size_t idx)
     {
-      y(idx) += *alpha * x(idx);
+        y(idx) += d_alpha * x(idx);
```
Comment (dalg24) [inline examples/03-axpy-view/axpy-view-cxx.cc]: Why did you change the indentation?

Comment (pkestene) [reply examples/03-axpy-view/axpy-view-cxx.cc]: ok, sorry, I restored the original indentation.
